### PR TITLE
Implicit function extensionality

### PR DESCRIPTION
**Pull request by fredrik-bakke**
Adds a series of basic lemmas for implicit functions related to function extensionality. Also adds type-theoretic principle of choice for them for better computation.

This is factored out from my other PRs, as I need it for all of them, so they all now depend on this one to be merged first. If I don't get any objections quickly, I will probably just merge this one without review.

### CODE REVIEW

**Comment by fredrik-bakke** [comment]
Just a note, I will review this PR myself before it is merged

File: src/foundation/implicit-function-types.lagda.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  equiv-implicit-explicit-Π : ((x : A) → B x) ≃ ({x : A} → B x)
+  pr1 equiv-implicit-explicit-Π = implicit-explicit-Π
+  pr2 equiv-implicit-explicit-Π = is-equiv-implicit-explicit-Π
+
+  equiv-explicit-implicit-Π : ({x : A} → B x) ≃ ((x : A) → B x)
+  pr1 equiv-explicit-implicit-Π = explicit-implicit-Π
+  pr2 equiv-explicit-implicit-Π = is-equiv-explicit-implicit-Π
+```
+
+### Equality if explicit functions is equality of implicit functions
```
Comment (VojtechStep) [inline src/foundation/implicit-function-types.lagda.md:56]:
```suggestion
### Equality of explicit functions is equality of implicit functions
```

File: src/foundation/type-theoretic-principle-of-choice.lagda.md [hunk 2]
```diff
@@ -30,26 +30,26 @@ distributivity of Π over Σ, is also known as the type theoretic principle of
 choice. Indeed, it is the Curry-Howard interpretation of (one formulation of)
 the axiom of choice.
 
-## Definitions
+## Distributivity of Π over Σ
```
Comment (VojtechStep) [inline src/foundation/type-theoretic-principle-of-choice.lagda.md:33]: I'd rather keep the file structure consistent with other files, i.e. have `Definitions > Distributivity of Pi over Sigma`, `Definitions > Distributivity or implicit Pi over Sigma`, `Properties > ...`; what motivated this change?

Comment (fredrik-bakke) [reply src/foundation/type-theoretic-principle-of-choice.lagda.md:33]: Yup, I agree the structure of this file is not yet good

Comment (fredrik-bakke) [reply src/foundation/type-theoretic-principle-of-choice.lagda.md:33]: The issue is I wanted to structure it as one part about explicit functions, and one about implicits. Each of them have a "statement" and "theorem" section.

Comment (fredrik-bakke) [reply src/foundation/type-theoretic-principle-of-choice.lagda.md:33]: Maybe the file can start with `## Theorem` and then `### Distributivity of Pi over Sigma`,  and then `#### Statement` and go from there

Comment (fredrik-bakke) [reply src/foundation/type-theoretic-principle-of-choice.lagda.md:33]: Eh, I guess having both statements at the start works fine. It's not like the definitions are very large.

**Comment by fredrik-bakke** [comment]
Thanks for the comments. Let me know if the new file structure is more to your liking

**Comment by VojtechStep** [comment]
Looks good, thanks 👌. I think I'd keep the `Properties` heading instead of replacing it with `Lemma`, since we usually put characterizing identity types under that, but it's not a hill I'm dying on, so feel free to merge

**Comment by fredrik-bakke** [comment]
In this instance, I think "lemma" is helpful to let readers know that the good stuff is further down in the file, so I'll merge it like this :)

**Pull request merged**